### PR TITLE
KAFKA-3817: handle null keys in KTableRepartitionMap

### PR DESCRIPTION
**Pull request by guozhangwang**

### CODE REVIEW

**Comment by guozhangwang** [comment]
@jklukas @enothereska @mjsax @dguy for reviews.

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableRepartitionMap.java [hunk 1]
```diff
@@ -66,32 +65,25 @@ public void enableSendingOldValues() {
...
+                throw new StreamsException("Record key for the grouping KTable should not be null.");
 
-            context().forward(newPair.key, new Change<>(newPair.value, null));
+            KeyValue<K1, V1> newPair = mapper.apply(key, change.newValue);
+            KeyValue<K1, V1> oldPair = mapper.apply(key, change.oldValue);
 
-            if (change.oldValue != null) {
-                KeyValue<K1, V1> oldPair = computeValue(key, change.oldValue);
+            // if the selected repartition key or value is null, skip
+            if (newPair.key != null && newPair.value != null) {
```
Comment (jklukas) [inline streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableRepartitionMap.java]: Should we check `newPair != null` before checking the key and value for nulls? A user might want to have their `mapper` return `null` in the case they want the value to be skipped.

Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableRepartitionMap.java]: Good point, ack.

**Comment by guozhangwang** [comment]
Addressed comments.

**Comment by jklukas** [comment]
LGTM +1

**Comment by guozhangwang** [comment]
Thanks @jklukas , merged to trunk and 0.10.0.

**Pull request closed without merging**